packages/mongo/collection/collection_extensions.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * Collection Extensions System
+ * 
+ * Provides a clean way to extend Mongo.Collection functionality
+ * without monkey patching. Supports constructor extensions,
+ * prototype methods, and static methods.
+ */
+
+CollectionExtensions = {
+  _extensions: [],
+  _prototypeMethods: new Map(),
+  _staticMethods: new Map(),
+  
+  /**
+   * Add a constructor extension function
+   * Extension function is called with (name, options) and 'this' bound to collection instance
+   */
+  addExtension(extension) {
+    if (typeof extension !== 'function') {
+      throw new Error('Extension must be a function');
+    }
+    this._extensions.push(extension);
+  },
+  
+  /**
+   * Add a prototype method to all collection instances
+   * Method is bound to the collection instance
+   */
+  addPrototypeMethod(name, method) {
+    if (typeof name !== 'string' || !name) {
+      throw new Error('Prototype method name must be a non-empty string');
+    }
+    if (typeof method !== 'function') {
+      throw new Error('Prototype method must be a function');
+    }
+    
+    // Check for reserved names (reserved for future hook system)
+    const reservedNames = ['before', 'after', 'direct', 'hookOptions', '_hooks'];
+    if (reservedNames.includes(name)) {
+      throw new Error(`Method name '${name}' is reserved for future hook system`);
+    }
+    
+    this._prototypeMethods.set(name, method);
```

Comment:
is there a reason for this(commit ref or something)?

This package could have in the future these hooks or am I missing something here?

Reply:
Nice! I'm _very_ happy with this effort!

Reply:
On that note having official `before` and `after` hooks for collection actions is something that we need to have in core.